PayButton/paybutton-server
# [#613] feat: payment triggers private key

Related to #613


Depends on
---
- [x] #631




Description
---
1. Adds a master secret key to the app and secret keys to each user.
2. Allows another variable on the trigger, `<postHash>`. This is the hash of the trigerring TX timestamp, using the user's secret key


Test plan
---

1. Add to your `.env.local` a `MASTER_SECRET_KEY=SOMESECRETKEY`. You can generate one with e.g. `openssl rand -base64 32`.
2. Create a trigger, e.g:
   postURL: `https://httpbin.org/post`
   postData:`{"Y": <postHash>, "t": <timestamp>, "id": <txId>}`

4. Then, after making a transaction, check in the db `SELECT * FROM TriggerLog;` and see something like: 
  `\"Y\":\"EXAMPLE_POST_HASH\", \"t\": 1693590786}`
in the postedData info.

5. Now go to your user Account page and get the secret key, e.g. `EXAMPLE_SECRET_KEY`

6. Finally, a user can check on his own that this post request is legit by executing e.g. in a bash command line:
```
echo -n "1693590786" | openssl dgst -sha256 -hmac "EXAMPLE_SECRET_KEY"
```

... and seeing if it's exactly `EXAMPLE_POST_HASH`

PS:  The command above can be executed inside the node container with `openssl1.1` instead of `openssl`
